### DISCUSSION TITLE
How to get the dimensions of the box around a GLTF/GLB model ?

### DISCUSSION THREAD

**Discussion by agGitHub** [sceneview/sceneform-android#243] [Q&A] [answered]
Hi,

I load 3D GLTF/GLB models with a CompletableFuture like [here](https://github.com/SceneView/sceneform-android/blob/master/samples/3d-model-viewer/src/main/java/com/google/ar/sceneform/samples/sceneviewbackground/MainActivity.java).

I would like to get the size of the box around the GLTF/GLB model. 

How can I do that ?

Thanks.

**Comment by grassydragon** [comment, marked as answer]
Hi!
You can access bounding box if you save a reference to `RendereableInstance` when setting `Renderable`. For example:
``` Kotlin
// Create a model node and fit it inside a unit cube
val modelNode = Node()
modelNode.parent = transformableNode

val instance = modelNode.setRenderable(state.model)
val boundingBox = instance.filamentAsset?.boundingBox

if (boundingBox != null) {
    val maxHalfExtent = boundingBox.halfExtent.maxOrNull()!!
    val scale = 0.5f / maxHalfExtent
    modelNode.localScale = Vector3(scale, scale, scale)
}
```

**Reply by Yeafun** [reply]
`// set scale
TransformableNode model = new TransformableNode(arFragment.getTransformationSystem());
     float scale = 0.1f;
     model.setLocalScale(new Vector3(scale, scale, scale));` if i use the TransformableNode, setLocalScale() method doesn't work.

**Reply by grassydragon** [reply]
That is right since the `TransformableNode` scale is controlled by the user. See how I'm using the `modelNode` as a child of the `TransformableNode`.
